### ISSUE TITLE
DATABASE: Populate plot.fvs_variant from PLOTGEOM.variant

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#318]
From @jsfried: Not sure how often PLOTGEOM is updated but out BioSum internal fiadb_fvs_variant table hasn’t been updated in probably 5-6 years (I never though to). I now see that it does need updating because FVS tweaks its maps a bit over time, apparently, and because new plots get added when spatial intensifications are undertaken (as they were in Oregon over the last few years). Relying on a corporate source for such changes (like PLOT GEOM) would leave fewer plots unaccounted for. Going forward, I think we should source variant from the FIADB table PLOTGEOM—one more table to pull into BioSum when adding plots, I guess. This is important now that BioSum is national. We will tackle this when/after we convert the plot table to sqlite

Optional:  Get the default move distance (fiadb_fvs_variant.MoveDistance_ft) as the midpoint of the road distance class in PLOT.RDDISTCD given that this distance is populated from the actual plot location, not from a point that can be up to .71 miles away.

![image](https://github.com/USFS-PNW/Fia-Biosum-Manager/assets/11527966/f2d1e7f8-61af-4e64-b95e-2f730dab4a68)

**Comment by lbross** [comment]
See issue #305. Note that if the analysis uses the option to update plot.gis_yard_dist_ft when loading the gis data, it will overwrite the value that was originally loaded from PLOT.RDDISTCD.

We are leaning towards populating plot.gis_yard_dist_ft from PLOT.RDDISTCD. @jsfried is working from a lookup table that he created from the FIADB definitions for PLOT.RDDISTCDs. @jsfried, can you provide this table when finalized so that we can append it to this issue and have it on hand when we're ready to implement this?

**Comment by jsfried** [comment]
[RD_DIST.csv](https://github.com/USFS-PNW/Fia-Biosum-Manager/files/15268297/RD_DIST.csv)
Here is the table. The codes are decoded to the upper limit, in feet, of the road distance class except for class 8 which is farther than 5 miles, which we code as -1. Users will need to be sure to set limits on the farthest distance to be considered in Optimizer for steep and not steep plots. 2640 or 5280 are recommended as maximum dist ance for ground based systems; probably 2640 for steep sloped systems. OpCost may behave badly with very large yarding distances.

**Comment by lbross** [comment]
1. PLOTGEOM table: Do we want to save a subset of these records, from the plots associated with the project, to the master.db? Or do we just extract the value that we need (fvs_variant) and set this value when the plots are loaded? 
2. Does this alleviate the need for the extra step we currently have to 'Plot FVS variants'? Or do we have to keep this step to populate the fvsloccode? Is there another source for fvsloccode than fiadb_fvs_variant?
We will need to add an extra field to the plot load screen to ensure that the PLOTGEOM table exists in the selected FIADB datamart.
3. Do we want to add a field to the PLOT table for RDDISTCD or is it enough that we populate the gis_yard_dist_ft using the matrix provided by @jsfried ?

**Comment by jsfried** [comment]
1. No need for PLOTGEOM table but DO obtain the values from PLOTGEOM into MASTER.PLOT (for FVS variant and LOC CODE)
2. Yes, we can rely on PLOTGEOM for both of these. This should allow us to skip the variant assignment step in BioSum (with the interface etc.). Yes, we will need to ensure that the PLOTGEOM table exists in FIADB datamart.
3. I prefer to add a field to MASTER.PLOT for RDDISTCD in case someone wishes to recode the field differently when crosswalking to a feet value in gis_yard_dist_ft (e.g., the mid point rather than the max distance for the class, as we are doing)

Fetching RDDIST class data from the current database is important as these change over time and any table we prepare today will be out of date soon. Consider this summary of WA data time 1 (2002-2011) and time 2 (2012-2021) field visits - the cells are the proportion of plots (considering only those that are at least partially forested at both visits and not access denied at either visit) for that time 1 and time 2 RDDISTCD assignment combination. While most of the time, the code does not change, in many cases, it does and in both directions.

![Image](https://github.com/user-attachments/assets/c02fd12d-7069-4e00-b5b3-d14f04631e3c)

**Comment by lbross** [comment]
From @jsfried: Yes, change the name. Consistency is good.

**Comment by lbross** [comment]
We need to store the matrix (RDDST_CD x DISTANCE_FT) in a table in the application. Original values provided earlier on this issue by @jsfried. Should this be in the appdata\biosum_ref.db where there is one copy for all projects and it can be overwritten by a new installation of BioSum? Or should it be in the project-specific ref_master.db where the analyst could customize it for each project. The starting values would come from BioSum. Also, name for the table? I suggest road_dist_ft

**Comment by jsfried** [comment]
Maybe road_class_2_feet_xwalk? project specific ref_master would be a more stable location, I think,

**Comment by lbross** [comment]
Project-specifc ref_master.db it is. I don't like numbers in field names. How about 'road_class_to_ft_xref'?

**Comment by jsfried** [comment]
That works-- I kind of expected some pushback on the "2" ;-) Also, I accidentally hit the unsubscribe button (instead of the view on github) for this thread and am unsure how to undo that inadvertent unsubscribe.

**Comment by jsfried** [comment]
Value in master.plot.gis_yard_dist_ft has changed-- used to be midpoint value for the RDDISTCD class which was correct (though incorrect values supplied by JF for classes 7 and 8 needed to be replaced with correct values). The field now appears to contain the max distance for the class, which is biased. There may have been discussion back and forth between LB and JF on this that briefly considered using Max rather than midpoint, but Max is not a good value to use as default. Correct midpoint value **MidDistFt** should populate here, NOT MaxDistFt or LUT_MidDistFt. Sorry for the trouble.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/jsfried/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/jsfried/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Code | Description | MinDistFt | MaxDistFt | MidDistFt | LUT_MidDistFt | Comment
-- | -- | -- | -- | -- | -- | --
1 | 100 feet or less | 0 | 100 | 50 | 50 |  
2 | 101 to 300 feet | 101 | 300 | 150 | 150 |  
3 | 301 to 500 feet | 301 | 500 | 400 | 400 |  
4 | 501 to 1000 feet | 501 | 1000 | 750 | 750 |  
5 | 1001 feet to 1/2 mile | 1001 | 2640 | 1820 | 1820 |  
6 | 1/2 to 1 mile | 2641 | 5280 | 3960 | 3960 |  
7 | 1 to 3 miles | 5281 | 15840 | 10560 | 10480 | this LUT value is incorrect; JF   error
8 | 3 to 5 miles | 15841 | 26400 | 21120 | 20960 | this LUT value is incorrect; JF   error
9 | Greater than 5 miles | 26401 | 100000 | 50000 | -1 | A large value like 50000 is   better and less confusing than a negative value



</body>

</html>

**Comment by lbross** [comment]
Good catch @jsfried. I have updated the values in ref_master.db\road_class_to_ft_xref. Will attach an image of the updated table below. Please review.
A deployment note: This table is copied into new projects and upgraded projects from the Program Files directory. This means that we don't have a way to automatically correct it for any projects created/upgraded from the initial v5.12.0 release. When we get the next .msi out, it will be an easy fix. You will just need to manually copy over the table again. But this is something that's easy to forget after we move past it.

<img width="205" height="265" alt="Image" src="https://github.com/user-attachments/assets/8d92c02a-3aab-46e3-bf87-87d8071939b7" />

**Comment by jsfried** [comment]
Thanks, Lesley, that looks great.

**Comment by lbross** [comment]
We are switching rddiscd 9 back to -1 as it will prevent having stands in tree vol val that aren't in harvest_costs. -1 prevents Processor from loading the stand and also lowers processing time.

**Comment by lbross** [comment]
Spot checked multiple plots (by PLOT) from new Klamath River Basin plots and confirmed that fvs_variant (and other variables) matched FIADB. Implemented in v5.12.0. Closing
